src/js/components/StatusBar.js

```diff
@@ -11,22 +11,74 @@ class StatusBar extends React.Component {
     if (max === 0) {
       return null;
     }
+    let indexesLessThanThreshold = [];
+    let maxIndex = 0;
+    let maxSize = 0;
+    // Size of largest portion to give to smallest portion. We have a max of 5
+    // status types so we want to keep this growth ratio relatively small.
+    let growRatio = 0.07;
+    // Grow anything below this threshold by stealing from the biggest portion
+    let threshold = 7;
+
+    let barSizes = data.map(function (status, index) {
+      let {value} = status;
+
+      let relativeSize = value / max * 100;
+
+      if (relativeSize > 0 && relativeSize < threshold) {
+        indexesLessThanThreshold.push(index);
+      }
+
+      return relativeSize;
+    });
+
+    function stealPortion() {
+      if (indexesLessThanThreshold.length) {
+        // Try to steal some width from the largest portion
+        // and give it to the first portion in indexesLessThanThreshold
+        maxSize = 0;
+        maxIndex = 0;
+        // Find new Max (could be different after we've stolen from a portion already)
+        barSizes.forEach(function (relativeSize, index) {
+          if (relativeSize > maxSize) {
+            maxSize = relativeSize;
+            maxIndex = index;
+          }
+        });
+
+        let sizeToSteal = maxSize * growRatio;
+
+        let currentGrowthIndex = indexesLessThanThreshold[0];
+        // Reassign the portion
+        barSizes[currentGrowthIndex] += sizeToSteal;
+        barSizes[maxIndex] -= sizeToSteal;
+
+        if (barSizes[currentGrowthIndex] > threshold) {
+          // Done with growing this portion. Otherwise it stays on the stack
+          // and will get another % added to it from the largest portion
+          // calculated in the next iteration.
+          indexesLessThanThreshold = indexesLessThanThreshold.slice(1);
+        }
+        // Recurse till we have grown everything above threshold
+        stealPortion();
+      }
+    }
+    stealPortion();
```

Comment:
What's the purpose of defining this function and calling it right after, wouldn't it just be enough with the if statement?